tools/metricq-discover.py

```diff
@@ -47,9 +56,118 @@
 click_completion.init()
 
 
+def camelcase_to_kebabcase(camelcase: str) -> str:
+    # Match empty string preceeding uppercase character, but not at the start
+    # of the word. Replace with '-' and make lowercase to get kebab-case word.
+    return re.sub(r"(?<!^)(?=[A-Z])", "-", camelcase).lower()
+
+
+class IgnoredEvent(Enum):
+    ErrorResponses = enum_auto()
+
+    @classmethod
+    def as_option_names(cls) -> List[str]:
+        return [camelcase_to_kebabcase(name) for name, _ in cls.__members__.items()]
+
+    @classmethod
+    def from_option_name(cls, option: str) -> "IgnoredEvent":
+        member_name = "".join(part.title() for part in option.split("-"))
+        return cls.__members__[member_name]
+
+
+class DiscoverResponse:
+    def __init__(
+        self,
+        alive: bool = True,
+        current_time: Optional[str] = None,
+        starting_time: Optional[str] = None,
+        uptime: Optional[int] = None,
+        metricq_version: Optional[str] = None,
+        hostname: Optional[str] = None,
+    ):
+        self.alive = alive
+        self.metricq_version = metricq_version
+        self.hostname = hostname
+
+        self.current_time = self._parse_datetime(current_time)
+        self.starting_time = self._parse_datetime(starting_time)
+        self.uptime: Optional[datetime.timedelta] = None
+
+        try:
+            if uptime is None:
+                self.uptime = self.current_time - self.starting_time
+            else:
+                self.uptime = (
+                    datetime.timedelta(seconds=uptime)
+                    if uptime < 1e9
+                    else datetime.timedelta(microseconds=int(uptime // 1e3))
+                )
+        except (ValueError, TypeError):
+            pass
+
+    @classmethod
+    def _parse_datetime(cls, iso_string) -> Optional[datetime.datetime]:
```

Comment:
I'm not quite sure if that should be in `metricq.Timestamp`. `parse_iso_datetime` comes from `dateutil` and is the only line here that acutally does something. The `dt.astimezone(tzlocal()).replace(tzinfo=None)` is just here because `humanize` cannot deal with non-UTC timezones ([yet](https://github.com/jmoiron/humanize/pull/144)).